src/pushprocessor.ts

```diff
@@ -96,6 +97,22 @@ const DEFAULT_OVERRIDE_RULES: IPushRule[] = [
             PushRuleActionName.DontNotify,
         ],
     },
+    {
+        // For homeservers which don't support MSC3786 yet
```

Comment:
I'm not sure this will work as you hope, maybe will only work in e2ee rooms, otherwise the server's notifications count will still be inflated if it doesn't support this MSC

Reply:
That is expected... (it probably isn't possible to get this working in a way we'd like with the current notification system...) This PR is analogues to https://github.com/matrix-org/matrix-js-sdk/pull/976: it expects the MSC implementation in Synapse to land, but in case your specific Synapse does not have it yet, it hopes to at least avoid the annoying notifications.

At least that is my understanding of why this code is here